compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -45,6 +46,18 @@ class InterceptedMethods extends MiniPhase {
       ctx.log(s"$phaseName rewrote $tree to $rewritten")
       rewritten
     }
+    else if (tree.name.startsWith(nme.UNARY_PREFIX.toString) && tree.qualifier.symbol.isAbsent) {
```

Comment:
I would swap the two conditions, the latter is probably less expensive.

Also why `isAbsent` instead of `!exists`? I think `isAbsent` is not fully equivalent to `!exists`  and carries some special meaning